### PR TITLE
firstMeaningfulPaint metric fixes

### PR DESCRIPTION
**Pull request by brendankenny**
taking over for #161

### CODE REVIEW

File: audits/performance/first-meaningful-paint.js [hunk 1]
```diff
@@ -71,17 +67,21 @@ class FirstMeaningfulPaint extends Audit {
...
+            duration: `${firstMeaningfulPaint.toFixed(2)}ms`,
             score: Math.round(score)
           };
-        }, _ => {
+        }, err => {
           // Recover from trace parsing failures.
-          return {
-            score: -1
-          };
+          if (err instanceof Error) {
```
Comment (samccone) [inline audits/performance/first-meaningful-paint.js]: what else could it be? :goat:

File: audits/performance/first-meaningful-paint.js [hunk 2]
```diff
@@ -71,17 +67,21 @@ class FirstMeaningfulPaint extends Audit {
           score = Math.max(0, score);
 
           return {
-            duration: `${fmp.duration.toFixed(2)}ms`,
+            duration: `${firstMeaningfulPaint.toFixed(2)}ms`,
             score: Math.round(score)
           };
-        }, _ => {
+        }, err => {
```
Comment (samccone) [inline audits/performance/first-meaningful-paint.js]: i really dislike the , style pipeline approach here... it makes it even easier to forget when you are in a catch and when you are not. Instead i prefer the explicit `.catch()` pipelining approah

Comment (brendankenny) [inline audits/performance/first-meaningful-paint.js]: done

File: audits/performance/first-meaningful-paint.js (hunk 1 above)
Comment (brendankenny) [inline audits/performance/first-meaningful-paint.js]: done

**Comment by paullewis** [comment]
Fabulous work, gents!

**Pull request merged**